community/kernel/src/main/java/org/neo4j/kernel/impl/store/CommonAbstractStore.java

```diff
@@ -163,37 +167,70 @@ public String getTypeDescriptor()
     }
 
     /**
-     * This method is called by constructors.
+     * This method is called by constructors. Checks the header record and loads the store.
+     * <p>
+     * Note: This method will map the file with the page cache. The store file must not
+     * be accessed directly until it has been unmapped - the store file must only be
+     * accessed through the page cache.
      * @param createIfNotExists If true, creates and initialises the store file if it does not exist already. If false,
      * this method will instead throw an exception in that situation.
      */
-    protected void checkStorage( boolean createIfNotExists )
+    protected void checkAndLoadStorage( boolean createIfNotExists )
     {
         int pageSize = pageCache.pageSize();
-        //noinspection EmptyTryBlock
-        try ( PagedFile ignore = pageCache.map( storageFileName, pageSize, ANY_PAGE_SIZE ) )
+        int filePageSize;
+        try ( PagedFile pagedFile = pageCache.map( storageFileName, pageSize, ANY_PAGE_SIZE ) )
         {
+            extractHeaderRecord( pagedFile );
+            filePageSize = pageCache.pageSize() - pageCache.pageSize() % getRecordSize();
         }
-        catch ( NoSuchFileException e )
+        catch ( NoSuchFileException | StoreNotFoundException e )
```

Comment:
My point was exactly that, when we "initialize new store file" we only touch the header record, and leave the rest of the file intact. There is no code that could delete or truncate a file. It is safe to do because it will never affect anything other than the header record.